lib/galaxy/workflow/modules.py

```diff
@@ -1799,7 +1791,9 @@ def from_workflow_step(Class, trans, step, **kwds):
                     new_tool_shed_url = new_url.split("/view")[0]
                     message += f'The tool \'{module.tool.name}\', version {tool_version} by the owner {module.tool.repository_owner} installed from <a href="{old_url}" target="_blank">{old_tool_shed_url}</a> is not available. '
                     message += f'A derivation of this tool installed from <a href="{new_url}" target="_blank">{new_tool_shed_url}</a> will be used instead. '
-            if step.tool_version and (step.tool_version != module.tool.version):
+            if step.tool_version and (
+                step.tool_version != module.tool.version and not get_safe_version(module.tool, step.tool_version)
```

Comment:
Should this be

```suggestion
                step.tool_version != module.tool.version and get_safe_version(module.tool, step.tool_version) != module.tool.version
```

?

Reply:
I don't think so, any returned version is fine, it doesn't need to match the tool version, it could be within the intermediate range.